File: LICENTA_GREENLIGHT/Server/greenlight/src/main/java/com/ciurezu/gheorghe/dragos/greenlight/data/entity/UserVoucher.java
package com.ciurezu.gheorghe.dragos.greenlight.data.entity;


import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;
import java.io.Serial;
import java.io.Serializable;

@Entity
@Table(name = "user_voucher")
@Setter
@Getter
public class UserVoucher implements Serializable {
    @Serial
    private static final long serialVersionUID = -5460812968266262395L;

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private Long Id;

    @ManyToOne
    @JoinColumn(name = "voucher_id", nullable = false)
    private Voucher voucher;

    @ManyToOne
    @JoinColumn(name = "users_id", nullable = false)
    private User user;

    @Column(name = "quantity")
    private Long quantity;
}

File: LICENTA_GREENLIGHT/Android/app/src/main/java/com/ciurezu/gheorghe/dragos/greenlight/data/network_model/get_voucher/UserVoucher.kt
package com.ciurezu.gheorghe.dragos.greenlight.data.network_model.get_voucher

import com.google.gson.annotations.SerializedName

data class UserVoucher(
    @SerializedName("voucher")
    val voucher: Voucher,
    @SerializedName("quantity")
    val quantity: Int
)
